### ISSUE TITLE
Support semantic indentation.

### ISSUE THREAD

**Issue by daewok** [sjlevine/atom-slime#39]
Slime has support for determining which forms are macro forms from the running lisp image and then indenting them properly.

https://common-lisp.net/project/slime/doc/html/Semantic-indentation.html

**Comment by haz** [comment]
This could be tricksy... @sjlevine -- would this mean completely overriding paredit for the formatting?

**Comment by sjlevine** [comment]
Yeah, I think paredit would need to change. I can imagine forking paredit, and making it call a function hook for each form to determine how much spacing to add. Then atom-slime could use a function hook that asks swank if it's a macro. 

But, I foresee trickery with JS promises...
